### PR TITLE
Update billable services workspace

### PR DESCRIPTION
**Pull request by agesa3**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Screenshots

https://github.com/user-attachments/assets/1f8bf417-3da0-44fa-a872-ecd53d4cb000

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
@agesa3 could you please attach a video showing what this PR does? Thanks

**Comment by agesa3** [comment]
> @agesa3 could you please attach a video showing what this PR does? Thanks



> @agesa3 could you please attach a video showing what this PR does? Thanks

I've attached a video but still there is an issue with the backend API when it comes to updating I had informed @Injiri about it and he was to action on it since the API creates a new entry rather than updating.If that is sorted out the code will be fully ready since on the frontend everything seems to be ok.

**Comment by ojwanganto** [comment]
@agesa3 - You may have forgotten to upload the video. Also, kindly resolve the conficts

**Comment by agesa3** [comment]
> @agesa3 - You may have forgotten to upload the video. Also, kindly resolve the conficts

Actioning on it

**Comment by ojwanganto** [comment]
Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.

**Comment by agesa3** [comment]
> Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.

Done.Updated the styling and video

> Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.

Done and updated the video.

@donaldkibet any guidance on this .In my local the import works ok but here on CI it seems the import of WorkspaceContainer is not found.
Screenshot 2024-07-23 at 11 24 09

File: packages/esm-billing-app/src/billable-services/billable-services.component.tsx [hunk 1]
```diff
@@ -37,6 +46,12 @@ const BillableServices = () => {
   const [showOverlay, setShowOverlay] = useState(false);
   const [overlayHeader, setOverlayTitle] = useState('');
 
+  const handleEditClick = (initialData) => {
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/billable-services.component.tsx:49]: Should we have a type for the `initialData` and probably update the name to something close e.g service

**Review by donaldkibet — APPROVED** [review]
Thanks @agesa3

**Comment by donaldkibet** [comment]
> > Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.
> 
> Done.Updated the styling and video
> 
> > Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.
> 
> Done and updated the video.
> 
> @donaldkibet any guidance on this .In my local the import works ok but here on CI it seems the import of WorkspaceContainer is not found. Screenshot 2024-07-23 at 11 24 09

We should update to the latest version of `@openmrs/esm-framework` to fix this error

**Comment by ojwanganto** [comment]
@agesa3 - thanks for the updates. You will at least need to refine styling of the form. Kindly ping the team once you have something to be reviewed.

**Comment by agesa3** [comment]
> @agesa3 - thanks for the updates. You will at least need to refine styling of the form. Kindly ping the team once you have something to be reviewed.

@ojwanganto sorry had uploaded the wrong video.Ive corrected and uploaded the correct one

**Comment by ojwanganto** [comment]
@agesa3 are you available to complete on this task? We need it urgently

**Comment by ojwanganto** [comment]
@agesa3 - so is the problem an issue with the backend?

**Comment by agesa3** [comment]
> @agesa3 - so is the problem an issue with the backend?

Yes, Simon had created a fix and it was merged and it worked ok but then a few days after that it started failing and whenever an update is made it creates a new entry rather than updating.

**Review by donaldkibet — APPROVED** [review]

**Pull request merged**